flocker/node/_deploy.py

```diff
@@ -62,7 +62,7 @@ class IDeployer(Interface):
     node_uuid = Attribute("")
     hostname = Attribute("")
 
-    def discover_state(cluster_state):
+    def discover_state(cluster_state, persistent_state):
```

Comment:
Perhaps the first argument should be `discovererable_state`? It wouldn't be unreasonable to have it as part of the first argument, except:
- their lifecycle is different: a `DeploymentState` is managed by `ClusterStateService` and updated by `IClusterStateChange` returned from discovery and is purely-ephemeral; `PersistentState` is managed by `ConfigurationPersistenceService` (perhaps now misnamed) and is persisted to disk.
- I want to say something about both objects being immutable and changing at different times, so there isn't an obvious place to create a composite object (other than perhaps here), but if that is the case, two arguments is pretty much equivalent.